source/common/router/upstream_request.cc

```diff
@@ -94,6 +94,20 @@ UpstreamRequest::~UpstreamRequest() {
         FilterUtility::percentageOfTimeout(response_time, parent_.timeout().per_try_timeout_));
   }
 
+  // Ditto for request/response size histograms.
+  Upstream::ClusterRequestResponseSizeStatsOptRef req_resp_stats_opt =
+      parent_.cluster()->requestResponseSizeStats();
+  if (req_resp_stats_opt.has_value()) {
```

Comment:
I think this will also correctly report bytes sent to multiple upstreams for hedging requests.  I think that's worth a regression test as well?

Reply:
The router creates one UpstreamRequest per each upstream request, so yeah we are covered. Yeah I can add something to the existing hedging tests or extend one of those.

Reply:
Done.